src/services/mentor.service.ts

```diff
@@ -0,0 +1,75 @@
+import { dataSource } from '../configs/dbConfig'
+import Mentor from '../entities/mentor.entity'
+import Profile from '../entities/profile.entity'
+import { ApplicationStatus } from '../enums'
+import Category from '../entities/category.entity'
+
+export const createMentor = async (req: any) => {
+  const userId = req.user.uuid
+  const userApplication = req.body
+
+  try {
+    const profileRepository = dataSource.getRepository(Profile)
+    const userProfile = await profileRepository.findOne({
+      where: { uuid: userId }
+    })
+
+    if (!userProfile) {
+      throw new Error('User profile not found')
+    }
+
+    const mentorRepository = dataSource.getRepository(Mentor)
+    const existingMentor = await mentorRepository.findOne({
+      where: { uuid: userId }
+    })
```

Comment:
There can be more than one mentor objects, ex: rejected and pending, rejected and approved